### ISSUE TITLE
MockTracer's activeSpan is not finished in try-resource block.

### ISSUE THREAD

**Issue by skrityak** [opentracing/opentracing-java#173]
The MockTracer example in https://github.com/opentracing/opentracing-java/tree/master/opentracing-mock does not work. 

The activeSpan created from MockTracer is NoopActiveSpan so calling deactivate()/close() by try-resource block does not finish the MockSpan .

**Comment by malafeev** [comment]
```
MockTracer tracer = new MockTracer();
``` 
should be replaced with 
```
MockTracer tracer = new MockTracer(new ThreadLocalActiveSpanSource())
```

**Comment by mabn** [comment]
Could it become the default? I was also surprised that `MockTracer` by default "does not work".

**Comment by pavolloffay** [comment]
+1 I have also seen people confused that MockTracer _does not work_ by default.

**Comment by malafeev** [comment]
fixed in #176
can be closed now.
